static/app/views/projectEventRedirect.tsx

```diff
@@ -24,7 +24,8 @@ function ProjectEventRedirect({router}: Props) {
     // This presumes that _this_ React view/route is only reachable at
     // /:org/:project/events/:eventId (the same URL which serves the ProjectEventRedirect
     // Django view).
-    const endpoint = router.location.pathname;
+    const [_, organization, ...rest] = router.location.pathname.split('/');
+    const endpoint = ['', 'organizations', organization, 'projects', ...rest].join('/');
```

Comment:
https://github.com/getsentry/sentry/blob/master/src/sentry/web/urls.py#L935-L940 is the new endpoint we're using here vs https://github.com/getsentry/sentry/blob/master/src/sentry/web/urls.py#L1231-L1235 is the old one.

Tested this by updating my local webpack.config.ts to proxy requests under `/organizations/` and rewriting the redirect location to my local domain with
```onProxyRes(proxyRes, req, res) {
  if (proxyRes.headers?.location?.includes('.sentry.io/issues/')) {
    proxyRes.headers.location = proxyRes.headers.location.replace(
      '.sentry.io/issues/',
      '.dev.getsentry.net:7999/issues/',
    );
  }
}
```

Reply:
this feels like it would be broken for people not using subdomains

Reply:
Could you explain how this may be broken?

Reply:
maybe i'm not able to parse the spread and join in my head. is there any way to use the router params or even helpers like useOrganization instead of parsing the pathname?

Reply:
Not sure that's trivial because the project slug is in the rest. The full route that we're parsing looks something like

```jsx
/:orgSlug/:projectSlug/events/:eventId
```
And we're splitting it into
```jsx
organization = :orgSlug
rest = :projectSlug/events/:eventId
```
And gets joined as
```jsx
/organizations/:orgSlug/projects/:projSlug/events/:eventId
```

Reply:
I'm not sure this is going to be a safe change. There are three UI routes that could resolve to this component

- `/:orgId/:projectId/events/:eventId/`
- `/projects/:projectId/events/:eventId/`
- `/organizations/:orgId/projects/:projectId/events/:eventId`

The string slicing won't work for all of the possible inbound routes.  We should be able to use `useParams()` to get the route template parameters. Those route parameters could be used to build the API endpoint URL safely.

Reply:
I'm not sure where
- `/projects/:projectId/events/:eventId/`
- `/organizations/:orgId/projects/:projectId/events/:eventId`
Could be coming from but I can definitely look into making sure they're supported. Assuming all 3 routes are handled, does this approach make sense?

Reply:
`/projects/:projectId/events/:eventId/` Is another legacy route that we have defined (but renders a broken page). And `/organizations/:orgId/projects/:projectId/events/:eventId/` gets picked up by a django view.